### DISCUSSION TITLE
SQL Trigger - compliance with general data protection rules

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#287] [Q&A] [answered]
In Europe / France we have rules on the conservation of personal data and in the framework of the software 
(https://en.wikipedia.org/wiki/General_Data_Protection_Regulation)

of the library:
- for the documents / loans : the delay is 45 days after the return of the document.
- for the member's account: the delay is 1 year after the last movement (contribution).

Can we have a trigger that launches the 2 constraints when BiblioteQ starts ?

Because I didn't see in the documentation any triggering element like the connection ?
INSERT / UPDATE / DELETE /

**Comment by textbrowser** [comment]
For PQ, reservation histories are optional for a patron. Of course, the database contains information about patrons and states of items. Anonymizing the relationships between patrons and items is not something BQ is responsible for.

Research anonymization techniques. Encrypted drives. Encrypted SQLite files. Encrypted PQ. These are some suggestions.

In another project, I work with encrypted data in regular databases. There are libraries which provide encryption for SQLite. The amount of work varies.

**Reply by meteos77** [reply]
In the loans we need to know the relationship : member - borrowed document.
However, we are not allowed to keep this information over time once the document has been returned.

but for the statistics we need to keep the borrowed document.

I work with a bash script
- save at the beginning permanently
- save at the end of the shift
- sqldiff START END --table item_borrower
then search in the database  with sqlite3 SELECT...
and I will test the creation of my statistics database.

**Comment by textbrowser** [comment]
PQ offers low-level security. This means that some entries may only be accessed by owners of the data. Suppose someone enables history. Another patron logging into the database (beyond BQ) can only read their entries. Administrators with correct privileges may read everyone.

**Comment by textbrowser** [comment]
There is published information for PQ security practices.

**Comment by meteos77** [comment]
If I understand correctly, PQ is likely to keep me busy for the whole year 2024 as there are so many options and features.

**Comment by textbrowser** [comment]
Encrypted drives, encrypted databases, permissions, good passwords, transport security - these are good things to have in PQ land. Some even have encrypted data. In BQ, patron information is private. Photographs and grey literature may be considered private. Of course you can upload encrypted data too.

**Comment by textbrowser** [comment]
BQ supported MySQL. I dropped it shortly after it was acquired by Oracle.

**Comment by textbrowser** [comment]
April, 2008.

**Comment by meteos77** [comment]
I have an openstreetmap project that goes with postgresql so I will have to spend some time on it. But for now it's the library and BQ.

geomatics an exciting project
https://static.geotribu.fr/articles/2022/2022-06-28_import-data-osm-postgresql-osm2pgsql-osmium/

**Reply by meteos77** [reply]
![news-pgAdmin4-v7 0](https://user-images.githubusercontent.com/56431039/234025526-114c1439-8a38-4949-9b96-9016c886eb41.png)

**Reply by meteos77** [reply]
In French, it's great

**Comment by textbrowser** [comment]
Connections to PQ may be global, local, both. You can restrict PQ so that it only listens and accepts connections on 127.0.0.1. Of course you can restrict addresses via iptables or ipfw or pf.

**Reply by meteos77** [reply]
For the library, no firewall problem -> we don't have internet :-)

**Comment by textbrowser** [comment, marked as answer]
Summary. In PostgreSQL, member reservation history is optional. In SQLite, member reservation history is optional in the newest release which is available in the repository as of the date of this comment.
